src/frontend/src/lib/types/listener.ts

```diff
@@ -1,5 +1,9 @@
+import type { Token } from '$lib/types/token';
+
 export interface WalletWorker {
 	start: () => void;
 	stop: () => void;
 	trigger: () => void;
 }
+
+export type InitWallerWorkerParams = { token: Token };
```

Comment:
nitpick: you can type the all function type, not just the params

```
type InitWalletWorkerFn = (params: {token: Token}) => Promise<WalletWorker>;

export const initWalletWorker: InitWalletWorkerFn = ({token}): : Promise<WalletWorker> => ...
```

Reply:
True, thanks 👍